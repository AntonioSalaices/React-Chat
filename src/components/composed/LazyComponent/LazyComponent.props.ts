export interface LazyComponentProps {
  /**
   * Children node
   */
  children: JSX.Element | JSX.Element[];
}
