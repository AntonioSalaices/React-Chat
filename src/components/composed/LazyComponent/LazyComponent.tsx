import { Suspense } from 'react';
import { PURPLE } from '@Utils/colors';
import { Spinner } from '@Components/Core';
import { LazyComponentProps } from './LazyComponent.props';
import { ErrorBoundary } from '../ErrorBoundary/ErrorBoundary';

const LazyComponent: React.FC<LazyComponentProps> = ({ children }) => {
  return (
    <Suspense
      fallback={
        <div>
          {' '}
          Loading... <Spinner color1={PURPLE} />{' '}
        </div>
      }
    >
      <ErrorBoundary>{children}</ErrorBoundary>
    </Suspense>
  );
};

export { LazyComponent };
