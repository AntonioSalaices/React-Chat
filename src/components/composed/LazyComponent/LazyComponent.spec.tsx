import { RenderResult, render } from '@testing-library/react';
import { LazyComponent } from './LazyComponent';

describe('<LazyComponent />', () => {
  let component: RenderResult;
  beforeEach(() => {
    component = render(
      <LazyComponent>
        <span>Hola</span>
      </LazyComponent>
    );
  });
  it('Should render lazy component and children', () => {
    const { container } = component;

    expect(container).toMatchSnapshot();
  });
});
