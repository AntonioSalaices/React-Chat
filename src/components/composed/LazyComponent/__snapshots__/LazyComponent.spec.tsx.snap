// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazyComponent /> Should render lazy component and children 1`] = `
<div>
  <span>
    Hola
  </span>
</div>
`;
